src/schema.ts

```diff
@@ -131,44 +164,44 @@ export function build(data: any, opt: QueryConfig = {}): Schema {
   return new Schema(fieldSchemas);
 }
 
-export class Schema {
+export class Schema implements TableSchema {
```

Comment:
in this scenario - what is the argument of TableSchema being parameterizable with `<FieldSchema>` ? In order to _be_ Table Schema, it needs to hold fields of a particular type. I believe the idea that `FieldSchema` **extends** `TableSchemaFieldDescriptor` would still work well in this encapsulated approach - and would simplify the signature of these interfaces.